virtual_rainforest/models/soil/soil_model.py

```diff
@@ -204,6 +218,7 @@ def integrate(self) -> dict[str, DataArray]:
                 self.data[str(name)].to_numpy()
                 for name in self.data.data.keys()
                 if str(name).startswith("soil_c_pool_")
+                or str(name).startswith("soil_enzyme_")
```

Comment:
You're only using `name` in stringified form here... maybe this will work instead?

```suggestion
                self.data[name].to_numpy()
                for name in map(str, self.data.data.keys())
                if name.startswith("soil_c_pool_")
                or name.startswith("soil_enzyme_")
```

Reply:
Ahh yes that works (i.e. `mypy` doesn't complain), and reads a lot more cleanly